core/services/ocr2/plugins/ccip/ccipexec/ocr2.go

```diff
@@ -387,15 +395,10 @@ func (r *ExecutionReportingPlugin) buildBatch(
 			continue
 		}
 
-		tokenData, err2 := getTokenData(ctx, msgLggr, msg, r.tokenDataProviders, skipTokenWithData)
-		if err2 != nil {
-			// When fetching token data, 3rd party API could hang or rate limit or fail due to any reason.
-			// If this happens, we skip all remaining msgs that require token data in this batch.
-			// If the issue is transient, then it is likely for other nodes in the DON to succeed and execute the msg anyway.
-			// If the issue is API outage or rate limit, then we should indeed avoid calling the API.
-			// If API issues do not resolve, eventually the root will only contain msg that should be skipped, and be snoozed.
-			skipTokenWithData = true
-			msgLggr.Errorw("Skipping message unable to get token data", "err", err2)
+		tokenData, dur, err := r.getTokenDataWithCappedLatency(ctx, msg, tokenDataRemainingDuration)
+		tokenDataRemainingDuration -= dur
```

Comment:
Would `dur` better be named `elapsed` to indicated the time passed while calling `getTokenDataWithCappedLatency`?

```suggestion
		tokenData, elapsed, err := r.getTokenDataWithCappedLatency(ctx, msg, tokenDataRemainingDuration)
		tokenDataRemainingDuration -= elapsed
```